### PR TITLE
Port  ocp4 mirror RPMs stage to Python

### PR DESCRIPTION
**Pull request by locriandev**
Ref. https://issues.redhat.com/browse/ART-3672

This is the first of many incremental changes by which we'll accomplish the full ocp4 porting. Since many locks and triggered job are involved, it felt safer to approach this task one step at the time. With this PR, `mirror RPMs` stage is ported to Python along with some groovy code currently in `commonlib`, that is used only by OCP4.

Test run: https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/ocp4/21/console
Synced plashets to mirror: https://s3.console.aws.amazon.com/s3/buckets/art-srv-enterprise?region=us-east-1&prefix=enterprise/all/4.7/latest/&showversions=false

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (Ximinhan) [inline pyartcd/pyartcd/pipelines/ocp4.py:40]: there will be a potential repomd.xml updated before previous cmd run complete

Comment (locriandev) [reply pyartcd/pyartcd/pipelines/ocp4.py:40]: The two `aws s3 sync` operations are sequential. We will `await` for the first one to complete before moving on to the next one.

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/ocp4.py:31]: I know the original groovy code doesn't do that, but could we add a check to ensure `s3_path` always starts with a `/`?

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/ocp4.py:11]: Why is `remove_old` parameter removed?

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/ocp4.py:83]: Let's also include stack traces in the error message: `traceback.format_exc()`

**Review by vfreex — COMMENTED** [review]
Thanks. I just commented on some minor issues. Otherwise this LGTM.

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/ocp4.py:32]: Done in cd4100ecee5272706a5b33cebd1ec880e3f65c1b

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/ocp4.py:85]: Done in 776ec1557ec71464c0d99faadfd5724840b152e3

File: pyartcd/pyartcd/pipelines/ocp4.py
Comment (locriandev) [reply pyartcd/pyartcd/pipelines/ocp4.py:12]: In our whole codebase, the only place where `remove_old` is set to False is within `sync-for-ci` job. That translates into a `aws s3 sync --delete` invocation that can be inlined whenever we bring that job to Python. Maybe we could move this function in a dedicated `s3.py` or the like file for reuse

Comment (vfreex) [reply pyartcd/pyartcd/pipelines/ocp4.py:12]: `remove_old` defaults to `True` and `aws s3 sync --delete` is invoked in the current groovy code.

Comment (locriandev) [reply pyartcd/pyartcd/pipelines/ocp4.py]: Thanks. Fixed in 28bce2ef7ef7fed49b69b64e40833945ba6430ef

**Review by vfreex — APPROVED** [review]
/lgtm

**Comment by locriandev** [comment]
/hold

waiting on next 4.13 RC before merging

**Pull request merged**
